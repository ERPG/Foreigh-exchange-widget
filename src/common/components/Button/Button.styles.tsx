import { makeStyles, Theme } from '@material-ui/core';

export const useButtonStyles = makeStyles((theme: Theme) => ({
  root: {},
}));
