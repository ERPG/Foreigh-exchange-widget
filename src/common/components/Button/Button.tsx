import * as React from 'react';
import { Button as MuiButton } from '@mui/material';

import { useButtonStyles } from './Button.styles';

export interface IButtonProps {
  disabled?: boolean;
  children: React.ReactNode;
  name: string;
  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const Button = ({
  name,
  disabled = false,
  children,
  onClick,
}: IButtonProps) => {
  const classes = useButtonStyles();

  return (
    <MuiButton
      fullWidth
      data-fx="fx-button"
      name={name}
      disabled={disabled}
      classes={{ root: classes.root }}
      disableElevation
      size="large"
      onClick={onClick}
      variant="contained"
    >
      <span>{children}</span>
    </MuiButton>
  );
};

export default Button;
