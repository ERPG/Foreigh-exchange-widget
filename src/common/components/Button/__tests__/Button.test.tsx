import { render, fireEvent } from '@testing-library/react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import Button from '../Button';

describe('Button component', () => {
  it('Should render component without errors', () => {
    const { container } = render(
      <Button name="test-button" onClick={() => ''}>
        Test button
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should handle click function', () => {
    const onClick = jest.fn();
    const { getByDataFx } = customRender(
      <Button name="test-button" onClick={onClick}>
        Test button
      </Button>
    );

    const buttonEl = getByDataFx('fx-button');
    expect(buttonEl).toBeDefined();

    fireEvent.click(buttonEl);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
