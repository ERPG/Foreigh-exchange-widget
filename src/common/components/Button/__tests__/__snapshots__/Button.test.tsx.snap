// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render component without errors 1`] = `
<div>
  <button
    class="makeStyles-root-1 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root css-1o1g6k0-MuiButtonBase-root-MuiButton-root"
    data-fx="fx-button"
    name="test-button"
    tabindex="0"
    type="button"
  >
    <span>
      Test button
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
